code/game/objects/structures/window.dm

```diff
@@ -78,6 +80,40 @@
 	if (flags_1 & ON_BORDER_1)
 		AddElement(/datum/element/connect_loc, loc_connections)
 
+/obj/structure/window/mouse_drop_receive(atom/dropping, mob/user, params)
+	. = ..()
+	if (added_leaning)
+		return
+	/// For performance reasons and to cut down on init times we are "lazy-loading" the leaning component when someone drags their sprite onto us, and then calling dragging code again to trigger the component
+	/// WALLSTATION EDIT BEGIN
+	var/offset = 11
+	if(!fulltile)
+		offset = 6
+	AddComponent(/datum/component/leanable, offset, same_turf = (flags_1 & ON_BORDER_1), lean_check = CALLBACK(src, PROC_REF(lean_check)))
+	added_leaning = TRUE
+	dropping.base_mouse_drop_handler(src, null, null, params)
+	/// comment: characters clip through directional windows if you're on the same tile so I'll just reduce the offset if it's directional
+	/// WALLSTATION EDIT END
+/obj/structure/window/proc/lean_check(mob/living/leaner, list/modifiers)
+	if (!(flags_1 & ON_BORDER_1))
+		return TRUE
+
+	if (leaner.loc == loc)
+		return dir == REVERSE_DIR(leaner.dir)
+
+	return get_dir(src, leaner) == dir && leaner.dir == dir
+
+/obj/structure/window/setDir(newdir)
```

Comment:
this proc should be probably somehow removed or made to do nothing, applies wallening offsets

Reply:
it should still call parent though